neurokit2/ecg/ecg_process.py

```diff
@@ -85,34 +89,31 @@ def ecg_process(ecg_signal, sampling_rate=1000, method="neurokit"):
       plt.close()
 
     """
-    # Sanitize input
-    ecg_signal = signal_sanitize(ecg_signal)
 
+    # Sanitize and clean input
+    ecg_signal = signal_sanitize(ecg_signal)
     ecg_cleaned = ecg_clean(ecg_signal, sampling_rate=sampling_rate, method=method)
-    # R-peaks
-    (instant_peaks, rpeaks,) = ecg_peaks(
-        ecg_cleaned=ecg_cleaned, sampling_rate=sampling_rate, method=method, correct_artifacts=True
-    )
 
+    # Detect R-peaks
+    instant_peaks, rpeaks = ecg_peaks(ecg_cleaned, sampling_rate, method, correct_artifacts=True)
+
+    # Calculate heart rate
     rate = signal_rate(rpeaks, sampling_rate=sampling_rate, desired_length=len(ecg_cleaned))
 
+    # Assess signal quality
     quality = ecg_quality(ecg_cleaned, rpeaks=rpeaks["ECG_R_Peaks"], sampling_rate=sampling_rate)
 
-    signals = pd.DataFrame(
-        {"ECG_Raw": ecg_signal, "ECG_Clean": ecg_cleaned, "ECG_Rate": rate, "ECG_Quality": quality}
-    )
+    # Merge signals in a DataFrame
+    signals = pd.DataFrame({"ECG_Raw": ecg_signal, "ECG_Clean": ecg_cleaned, "ECG_Rate": rate, "ECG_Quality": quality})
 
-    # Additional info of the ecg signal
-    delineate_signal, delineate_info = ecg_delineate(
-        ecg_cleaned=ecg_cleaned, rpeaks=rpeaks, sampling_rate=sampling_rate
-    )
+    # Delineate QRS complex
+    delineate_signal, delineate_info = ecg_delineate(ecg_cleaned, rpeaks, sampling_rate)
 
-    cardiac_phase = ecg_phase(ecg_cleaned=ecg_cleaned, rpeaks=rpeaks, delineate_info=delineate_info)
+    # Determine cardiac phases
+    cardiac_phase = ecg_phase(ecg_cleaned, rpeaks, delineate_info)
 
+    # Add additional information to signals DataFrame
     signals = pd.concat([signals, instant_peaks, delineate_signal, cardiac_phase], axis=1)
 
-    # Rpeaks location and sampling rate in dict info
-    info = rpeaks
-    info["sampling_rate"] = sampling_rate
```

Comment:
I would be wary of removing information from this dictionary - any justification here?

Reply:
Sure, the `sampling_rate` should be in there. But `rpeaks` is returned by `ecg_peaks()` and already contains the `sampling_rate` (compare line 265 in `ecg_peaks.py`). So it is just set again here with the same value. I removed this part to make it more easily comprehensible. External behaviour does not change.

Reply:
Ah I see your point, good catch & thanks for the explanation!